### PR TITLE
filtering alerts by state and display count of alerts in each state

### PR DESCRIPTION
**Pull request by mszczygiel**
This is improvement I made on https://github.com/prometheus/prometheus/issues/5655
In my first PR in this area (https://github.com/prometheus/prometheus/pull/5664) someone noted, that simply displaying active alerts on top would make order basically random so it would be hard to track single group. In this PR I took different approach - Added possibility to filter alerts and also number of alerts in each state is displayed, as on below screenshot
2019-07-12-123518_572x380_scrot

I hope this will make more people happy than shuffling alerts.

### CODE REVIEW

**Comment by kustodian** [comment]
If any of the filters is selected and if a particular group doesn't have alerts which are in the filtered states, those groups should be left empty. We have 50+ groups and if empty groups are displayed it will look really bad and could hide alerts which you want to show using filters.

**Comment by mszczygiel** [comment]
Sure, I can hide groups without alerts displayed. Would be good to have any opinion from maintainers what they prefer / what's likely to be merged.

**Comment by ilicmilan** [comment]
What's going on with this PR?
Travis test failed

**Comment by mszczygiel** [comment]
I think the reason why it failed is I force pushed sing-off commit too fast.

**Comment by simonpasquier** [comment]
@mszczygiel you need to rebase your PR on top of master.

File: web/ui/static/js/alerts.js [hunk 1]
```diff
@@ -26,6 +26,47 @@ function init() {
         targetEl.removeClass('is-checked');
     }
   });
+
+  function displayAlerts(alertState, shouldDisplay) {
```
Comment (zerkms) [inline web/ui/static/js/alerts.js]: Could this whole function be rewritten to

```js
  function displayAlerts(alertState, shouldDisplay) {
      $("#alertsTable > tbody > tr." + alertState).each(function(_, container) {
          $(container).toggle(shouldDisplay);
      });
  }
```

?

Comment (mszczygiel) [reply web/ui/static/js/alerts.js]: sure, thanks!

**Comment by mszczygiel** [comment]
@simonpasquier I've just rebased it.

File: web/ui/static/js/targets.js [hunk 2]
```diff
@@ -22,18 +22,17 @@ function showUnhealthy(_, container) {
...
-    } else if (localStorage.selectedTab == "unhealthy-targets") {
+    } else if (localStorage.selectedTargetsTab == "unhealthy-targets") {
       $("#unhealthy-targets").parent().addClass("active");
       $(".table-container").each(showUnhealthy);
     }
   } else {
     $(".table-container").each(showAll);
   }
 
-  $("button.targets").click(function() {
```
Comment (simonpasquier) [inline web/ui/static/js/targets.js]: hmm unexpected deletion?

Comment (mszczygiel) [reply web/ui/static/js/targets.js]: yes, a mistake during rebase. Thanks, fixed

**Comment by simonpasquier** [comment]
@mszczygiel would you have time to update the PR?

**Comment by mszczygiel** [comment]
@simonpasquier sure, I'll apply fixes later this week.

File: web/ui/static/js/alerts.js [hunk 3]
```diff
@@ -26,6 +26,41 @@ function init() {
...
+  if(localStorage.hideFiringAlerts === "true") {
+      $("#firingAlerts").parent().removeClass("active");
+      displayAlerts("alert-danger", false);
+  }
+
+  $("#alertFilters :input").change(function() {
+        const target = $(this).attr("id");
+        var isActive = $(this).parent().hasClass("active");
+        if (target === "inactiveAlerts") {
+            localStorage.setItem("hideInactiveAlerts", isActive);
```
[the remark this answers is not in the document]
Comment (mszczygiel) [reply web/ui/static/js/alerts.js]: ok, done

**Comment by druss17** [comment]
Any idea when this PR will be merged?

**Comment by juliusv** [comment]
@simonpasquier Thanks. We're still a bit away from implementing the non-graph pages in React, but by that time we'll take a close look at all the existing JS for those and convert it.

**Comment by simonpasquier** [comment]
Thanks!

**Pull request merged**